x-pack/plugins/ai_infra/product_doc_base/server/saved_objects/product_doc_install.ts

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { SavedObjectsType } from '@kbn/core/server';
+import type { ProductName } from '@kbn/product-doc-common';
+import { productDocInstallStatusSavedObjectTypeName } from '../../common/consts';
+import type { InstallationStatus } from '../../common/install_status';
+
+/**
+ * Interface describing the raw attributes of the product doc install SO type.
+ * Contains more fields than the mappings, which only list
+ * indexed fields.
+ */
+export interface ProductDocInstallStatusAttributes {
+  product_name: ProductName;
+  product_version: string;
+  installation_status: InstallationStatus;
+  last_installation_date?: number;
+  last_installation_failure_reason?: string;
+  index_name?: string;
+}
+
+export const productDocInstallStatusSavedObjectType: SavedObjectsType<ProductDocInstallStatusAttributes> =
+  {
+    name: productDocInstallStatusSavedObjectTypeName,
+    hidden: true,
+    namespaceType: 'agnostic',
```

Comment:
I sat thinking about this one for quite a while. I think I agree with your approach, but I know other teams have been bitten by agnostic types in the past when they eventually discover a need for space-awareness (👋 @shahzad31)

Reply:
Yeah, I was the Core member working on that problematic, I think I know the risks and drawbacks 😅. Which is why I think that we do want install status (is the index/data here or not) space agnostic (doing that with a multi-space SO would have 99 problems when trying to see if the doc is installed from a space it's not accessible yet). And if we eventually need to make the feature space-aware later, it will have to be done via an additional, space aware, SO type (1-n relation)